plugin/pkg/scheduler/algorithm/priorities/util/non_zero.go

```diff
@@ -48,3 +53,68 @@ func GetNonzeroRequests(requests *api.ResourceList) (int64, int64) {
 	}
 	return outMilliCPU, outMemory
 }
+
+// FilterPodsByNameSpaces filters the pods based the given list of namespaces
+func FilterPodsByNameSpaces(names sets.String, pods []*api.Pod) []*api.Pod {
+	if len(pods) == 0 || len(names) == 0 {
+		return pods
+	}
+	result := []*api.Pod{}
+	for _, pod := range pods {
+		if names.Has(pod.Namespace) {
+			result = append(result, pod)
+		}
+	}
+	return result
+}
+
+// GetNamespacesFromPodAffinityTerm returns a set of names
+// according to the namespaces indicated in podAffinityTerm.
+// if the NameSpaces is nil considers the given pod's namespace
+// if the Namespaces is empty list then considers all the namespaces
+func GetNamespacesFromPodAffinityTerm(pod *api.Pod, podAffinityTerm api.PodAffinityTerm) sets.String {
+	names := sets.String{}
+	if podAffinityTerm.Namespaces == nil {
+		names.Insert(pod.Namespace)
+	} else if len(podAffinityTerm.Namespaces) != 0 {
+		for _, nameSpace := range podAffinityTerm.Namespaces {
+			names.Insert(nameSpace.Name)
+		}
+	}
+	return names
+}
+
+// NodeHasTopologyKey checks if nodeA and nodeB have same label value with given topologyKey as label key.
+// If the topologyKey is nil/empty, regard as the two nodes have same topologykey and label value.
+func NodesHaveSameTopologyKey(nodeA *api.Node, nodeB *api.Node, topologyKey string) bool {
+	if len(topologyKey) == 0 {
```

Comment:
> Sorry, the design doc under-specified what was intended for empty topologyKey. The use case was specifically for soft anti-affinity, so a user could say "spread across all topologies" without having to know the names of the topologies, and to make it easy for the administrator to change the set of topologies without users having to change their configs. "All topologies" doesn't really make sense for affinity, or for hard anti-affinity, which is why I suggested in my comment on TopologyKey in types.go that you should have validation reject empty TopologyKey when it's used with PodAffinity and when it's used with hard PodAntiAffinity.

To be clear, the semantic "all topologies", do you mean:
- When topology-key is empty, it actually means to match all the label keys from `--failure-domains`(no matter they are default values or user inputs), but not to match all the label keys exist on the node.

Correct me if I missed anything.